runners/extensions-java/metrics/src/main/java/org/apache/beam/runners/extensions/metrics/MetricsGraphiteSink.java

```diff
@@ -0,0 +1,305 @@
+              distribution.getCommitted().getSum(),
+              metricTimestamp);
+      messagePayload.append(metricMessage);
+      metricMessage =
+          String.format(
+              Locale.US,
+              "%s %s %s\n",
+              createNormalizedMetricName(
+                  distribution, "distribution", "mean", CommittedOrAttemped.COMMITTED),
+              distribution.getCommitted().getMean(),
+              metricTimestamp);
+      messagePayload.append(metricMessage);
+    } else {
+      metricMessage =
+          String.format(
+              Locale.US,
+              "%s %s %s\n",
+              createNormalizedMetricName(
+                  distribution, "distribution", "min", CommittedOrAttemped.ATTEMPTED),
+              distribution.getAttempted().getMin(),
+              metricTimestamp);
+      messagePayload.append(metricMessage);
+      metricMessage =
+          String.format(
+              Locale.US,
+              "%s %s %s\n",
+              createNormalizedMetricName(
+                  distribution, "distribution", "max", CommittedOrAttemped.ATTEMPTED),
+              distribution.getAttempted().getMax(),
+              metricTimestamp);
+      messagePayload.append(metricMessage);
+      metricMessage =
+          String.format(
+              Locale.US,
+              "%s %s %s\n",
+              createNormalizedMetricName(
+                  distribution, "distribution", "count", CommittedOrAttemped.ATTEMPTED),
+              distribution.getAttempted().getCount(),
+              metricTimestamp);
+      messagePayload.append(metricMessage);
+      metricMessage =
+          String.format(
+              Locale.US,
+              "%s %s %s\n",
+              createNormalizedMetricName(
+                  distribution, "distribution", "sum", CommittedOrAttemped.ATTEMPTED),
+              distribution.getAttempted().getSum(),
+              metricTimestamp);
+      messagePayload.append(metricMessage);
+      metricMessage =
+          String.format(
+              Locale.US,
+              "%s %s %s\n",
+              createNormalizedMetricName(
+                  distribution, "distribution", "mean", CommittedOrAttemped.ATTEMPTED),
+              distribution.getAttempted().getMean(),
+              metricTimestamp);
+      messagePayload.append(metricMessage);
+    }
+    return messagePayload.toString();
+  }
+
+  private <T> String createNormalizedMetricName(
+      MetricResult<T> metric,
+      String metricType,
+      String valueType,
+      CommittedOrAttemped committedOrAttemped) {
+    String metricName =
+        String.format(
+            "beam.%s.%s.%s.%s.%s",
+            metricType,
+            metric.getName().getNamespace(),
+            metric.getName().getName(),
+            committedOrAttemped,
+            valueType);
+
+    return WHITESPACE.matcher(metricName).replaceAll(SPACE_REPLACEMENT);
+  }
+
+  private enum CommittedOrAttemped {
```

Comment:
no, only two type of metrics

Reply:
I was asking because these status names are incorporated into enum name, so it will be difficult to add another one if it's needed. I suggest to rename it with more generic name, like `MetricStatus` or similar one.

Reply:
Take a look at this class: `AttemptedAndCommitted` in the SDK. I did this out of consistency, commited and attempted are the 2 types defined at beam model level.

Reply:
Ok, let's keep it so (though, it' not a good practice in my opinion).